src/tools/getBuildFailureLogs/handler.ts

```diff
@@ -27,43 +22,32 @@ export const getBuildFailureLogs: ToolCallback<{
   let projectSlug: string | null | undefined;
   let pipelineNumber: number | undefined;
 
-  if (failedPipelineURL || failedJobURL) {
-    const failedURL = (failedPipelineURL ?? failedJobURL)!;
-    projectSlug = getProjectSlugFromURL(failedURL);
-    pipelineNumber = parseInt(getPipelineNumberFromURL(failedURL));
+  if (projectURL) {
+    projectSlug = getProjectSlugFromURL(projectURL);
+    try {
+      pipelineNumber = getPipelineNumberFromURL(projectURL);
+    } catch {
+      pipelineNumber = undefined;
+    }
```

Comment:
`getPipelineNumberFromURL` will throw if it can't find the pipelineNumber from the url or if it isn't a number. so we can set `pipelineNumber = undefined` when that happens. and that will play nice with how it is used down below:

```ts
  const logs = await getPipelineJobLogs({
    projectSlug,
    branch,
    pipelineNumber,
  });
```